indra/llui/lluictrl.cpp

```diff
@@ -413,13 +416,26 @@ BOOL LLUICtrl::handleDoubleClick(S32 x, S32 y, MASK mask)
 	return handled;
 }
 
+// virtual
+BOOL LLUICtrl::handleDragAndDrop(S32 x, S32 y, MASK mask, BOOL drop, EDragAndDropType cargo_type,
+    void* cargo_data, EAcceptance* accept, std::string& tooltip_msg)
+{
+    if (mDragAndDropSignal)
```

Comment:
This feel out of order, with this implementation if a tab handles drag an drop this way any child (like inventory in said tab) would not get the drop signal. I think it should be like

res = LLView::handleDragAndDrop(...);
if (!res && mDragAndDropSignal)
{
   return (*mDragAndDropSignal)(...)
}

Reply:
I believe this order allows the object to stop the inherited processing if needed.
In case when the control wants to continue inherited processing it returns false.

If we call inherited processing first, the control can't stop this processing.
So, I still mean the proposed order to be right:
specific processing goes first and controls the execution of default processing.

Reply:
By default it is normal practice to let children handle drag an drop, because they are in the front and that's what LLView::handleDragAndDrop does.

Children are in the front they must have first dibs by default.